src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -428,6 +434,28 @@ private static void ConfigureCosmosDBInfrastructure(AzureResourceInfrastructure
     }
 }
 
+/// <remarks>
+/// The provisioning APIs will mark the DisableLocalAuth as `ReadOnly` after the
+/// `IsExistingResource` property is set, making it impossible to configure the
+/// `DisableLocalAuth` property in our usual flows. This is a workaround to
+/// allow us to set the property on existing resources.
+/// </remarks>
+internal sealed class ExistingCosmosDBAccount : CosmosDBAccount
+{
+    public ExistingCosmosDBAccount(string bicepIdentifier, bool disableLocalAuth, string? resourceVersion = null) : base(bicepIdentifier, resourceVersion)
+    {
+        // only explicitly set DisableLocalAuth if WithAccessKeyAuthentication was called.
+        // We don't want to disable local auth on an existing resource if it is already enabled
+        // as that might break other uses of the resource.
+        if (disableLocalAuth is false)
+        {
+            DisableLocalAuth = false;
```

Comment:
@captainsafia - I see for Redis that we always set it, is that intentional?

https://github.com/dotnet/aspire/blob/acdf5012407c2ab0259f4b6f075acb6c668949f3/src/Aspire.Hosting.Azure.Redis/AzureRedisExtensions.cs#L287-L294

https://github.com/dotnet/aspire/blob/acdf5012407c2ab0259f4b6f075acb6c668949f3/src/Aspire.Hosting.Azure.Redis/AzureRedisExtensions.cs#L205-L208

Reply:
Yes, this was intentional although you make a good point in your comment here and we probably want to constrain the Redis resource in the same way.

Reply:
It turns out this doesn't work end-to-end. If you set this property on an existing resource you get:

> Error BCP173: The property "properties" cannot be used in an existing resource declaration. [https://aka.ms/bicep/core-diagnostics#BCP173]

See https://github.com/Azure/azure-sdk-for-net/issues/48352#issuecomment-2669386450

Reply:
I've updated CosmosDB and Redis to no longer do this. I will make a follow up for Postgres since it has the same problem.